core/tests/protractor_desktop/collections.js

```diff
@@ -156,12 +162,16 @@ describe('Collections', function() {
     collectionEditorPage.setObjective('This is a test collection.');
     collectionEditorPage.setCategory('Algebra');
     collectionEditorPage.saveChanges();
+    waitFor.invisibilityOf(
+      saveInProgressLabel, 'Collection is taking too long to save.');
     users.logout();
   });
 
   it('visits the collection player', function() {
     users.login('alice@collections.com');
-    browser.get('/collection/' + collectionId);
+    libraryPage.get();
+    libraryPage.findCollection('Test Collection');
```

Comment:
Can we follow this pattern in other places in this file and remove collectionID var? (Just wanted to check if that's possible else I'm fine with doing this separately.)

Reply:
I tried to get rid of browser.getCurrentUrl() but that didn't work as expected.

Reply:
Just wanted to check, what issue you faced (any log or maybe you can explain)? (Might be helpful for us during re-structuring the e2etests.)

Reply:
I do not have any logs for it. It was more of a design issue than an error. My approach was to try to navigate to the appropriate editor page or the player page and access the anchor tag url and parse that for the id. However, it turns out that the protractor way of doing this is basically an async method that resolves with a string/null (see [getAttribute](https://www.protractortest.org/#/api?view=webdriver.WebElement.prototype.getAttribute)). To go forward with this, we would have to put all the code following this "id getter" method inside the "then" portion of it. It would end up looking like the code in the previous test (see [L66](https://github.com/oppia/oppia/pull/8900/files#diff-4989cfcd8c095c5483e9bf7797221ca4R66)), and that didn't seem like the proper way to go. So I decided against it.